src/Agent.Plugins/BuildDrop/BuildDropDropPlugin.cs

```diff
@@ -0,0 +1,234 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.TeamFoundation.Build.WebApi;
+using Microsoft.VisualStudio.Services.BlobStore.Common;
+using Microsoft.VisualStudio.Services.Content.Common.Tracing;
+using Microsoft.VisualStudio.Services.BlobStore.WebApi;
+using Microsoft.VisualStudio.Services.Common;
+using Microsoft.VisualStudio.Services.WebApi;
+using Microsoft.TeamFoundation.DistributedTask.WebApi;
+using Microsoft.VisualStudio.Services.Agent.Util;
+using Agent.Sdk;
+
+namespace Agent.Plugins.BuildDrop
+{
+    public abstract class BuildDropTaskPluginBase : IAgentTaskPlugin
```

Comment:
i am not sure whether we need to have a abstract class, since 5/6 methods/feilds are abstract.
I would suggest just have BuildDropUploadTask:IagentTaskPlugin and BuildDropDonwloadTask: IAgentTaskPlugin

Reply:
the reason for RepositoryPlugin has an abstract base class is  checkout is one task that has `main` and `post` stage.

Reply:
I refactored this a little bit to reduce complexity. There are about four arguments which are shared between the two tasks so I think it's a good idea to let RunAsync in abstract lass handle that commonality.